lib/mutgen.c

```diff
@@ -25,6 +25,18 @@
 
 #include "msprime.h"
 
+typedef struct {
+    tsk_mutation_t *mutation;
+    double time;
+    struct mutation_list_t *next;
+} mutation_list_t;
+
+typedef struct {
+    tsk_site_t *site;
+    mutation_list_t *mutations_head;
+    tsk_size_t mutations_length;
+} site_list_t;
```

Comment:
Sorry, I misread this the first time around. This isn't a linked list of sites, it's a wrapper for a tsk_site_t and a list of mutations. We could just call it ``site_t``? Also simpler to embed the tsk_site_t in there rather than a pointer, i.e.,
```
typedef struct {
￼    tsk_site_t site;
￼    mutation_list_t *mutations_head;
￼    tsk_size_t mutations_length;
￼} site_t;
```

That way we don't have to malloc the tsk_site individually.